filecopy/operations/services/notification/client.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2022 Indoc Research
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation, either version 3 of the License, or any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along with this program.
+# If not, see http://www.gnu.org/licenses/.
+
+import logging
+from typing import List
+from typing import Optional
+
+from operations.models import Node
+from operations.models import NodeList
+from operations.services.notification.models import InvolvementType
+from operations.services.notification.models import Location
+from operations.services.notification.models import NotificationType
+from operations.services.notification.models import PipelineAction
+from operations.services.notification.models import PipelineNotification
+from operations.services.notification.models import PipelineStatus
+from operations.services.notification.models import Target
+from requests import Session
+
+logger = logging.getLogger(__name__)
+
+
+class NotificationServiceClient:
+    """Client for sending notifications into Notification Service."""
+
+    def __init__(
+        self,
+        endpoint: str,
+        include_nodes: NodeList,
+        source_folder: Node,
+        destination_folder: Optional[Node],
+        project_code: str,
+        pipeline_action: PipelineAction,
+        pipeline_status: PipelineStatus,
+        operator: str,
+        notification_type: NotificationType,
+    ) -> None:
+        self.endpoint = f'{endpoint}/v1/all/notifications/'
+        self.include_nodes = include_nodes
+        self.source_folder = source_folder
+        self.destination_folder = destination_folder
+        self.project_code = project_code
+        self.pipeline_action = pipeline_action
+        self.pipeline_status = pipeline_status
+        self.operator = operator
+        self.notification_type = notification_type
+        self.client = Session()
+
+    def set_status(self, status: str) -> None:
+        self.pipeline_status = status
+
+    def set_location(self, entity: Node) -> Location:
+        return Location(id=entity.id, path=str(entity.display_path), zone=entity.zone).dict()
+
+    def set_targets(self) -> List[dict]:
+        targets = []
+        for node in self.include_nodes:
+            file_node = self.include_nodes[node]
```

Comment:
Since it's a dict here we can simplify:
```suggestion
        for node, file_node in self.include_nodes.items():
```